### PR TITLE
#55 implement endpoint requete groupe status

### PR DESCRIPTION
**Pull request by Chombon**

### CODE REVIEW

File: backend/src/main/java/com/ambacam/service/RequeteService.java [hunk 1]
```diff
@@ -291,6 +297,53 @@ public Requete updateStatus(Long id, RequeteStatusTO requeteStatusTO) {
...
+	    // find requete groupe
+	    RequeteGroupe requeteGroupe = findRequeteGroupe(requeteGroupeId);
+	    
+	    // find status
+	    StatusRequete status = findStatusRequete(assignStatusTO.getStatusId());
+	    
+	    // find requetes
+	    List<Requete> requetes = requeteRepository.findAllByRequeteGroupe(requeteGroupe);
+	    
+	    assignStatusTO.getRequeteIds().forEach(requeteId -> {
```
Comment (MentalistHB) [inline backend/src/main/java/com/ambacam/service/RequeteService.java]: Maybe the swagger was understood wrong. The transfer object here is a RequeteStatusTO, which is just about transporting the status id, all the requests belonging to the group should be assigned with.
From that point of view, I believe that the implementation here should just update the requests belonging to this requeteGroupe with the specified status.
Please take a look at the swagger again.

Is there any reason why you implemented it that way?

I believe that the function responsible for that should be in the `RequeteGroupeService` and call the function `RequeteService.updateStatus()`.

File: backend/src/main/java/com/ambacam/transfert/requetes/AssignStatusTO.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
+package com.ambacam.transfert.requetes;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import javax.validation.constraints.NotNull;
+
+import org.hibernate.validator.constraints.NotEmpty;
+
+public class AssignStatusTO {
```
Comment (MentalistHB) [inline backend/src/main/java/com/ambacam/transfert/requetes/AssignStatusTO.java]: Regarding the comment up there, this class would become obsolete after a discussion.

File: backend/src/main/java/com/ambacam/service/RequeteGroupeService.java [hunk 3]
```diff
@@ -204,6 +205,57 @@ public RequeteGroupeReadTO updateStatus(Long requeteGroupeId, IdTO<Long> requete
...
+     * Assign/remove requetes to/from a requete groupe
+     * 
+     * @param id
+     * @param addRemoveTO
+     * 
+     * @throws ResourceNotFoundException
+     *             if the requeteGroupe is not found
+     * @throws ResourceNotFoundException
+     *             if a requete does not exist
+     * @throws ResourceNotFoundException
```
[the remark this answers is not in the document]
Comment (Chombon) [reply backend/src/main/java/com/ambacam/service/RequeteGroupeService.java]: Yes the description is wrong

File: backend/src/main/java/com/ambacam/service/RequeteService.java (hunk 1 above)
Comment (Chombon) [reply backend/src/main/java/com/ambacam/service/RequeteService.java]: This function is the implementation of the service for the endpoint: 
"/operateurs/{operateurId}/requete-groupes/{requeteGroupeId}/requetes/status"
Please take a look at the swagger regarding this endpoint. Maybe the specification is wrong.

File: backend/src/main/java/com/ambacam/service/RequeteGroupeService.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (Chombon) [reply backend/src/main/java/com/ambacam/service/RequeteGroupeService.java]: Yes the description is wrong. I made a mistake in the description

File: backend/src/main/java/com/ambacam/service/RequeteService.java (hunk 1 above)
Comment (MentalistHB) [reply backend/src/main/java/com/ambacam/service/RequeteService.java]: You are right! Thanks for that.

File: backend/src/main/java/com/ambacam/transfert/requetes/AssignStatusTO.java (hunk 2 above)
Comment (MentalistHB) [reply backend/src/main/java/com/ambacam/transfert/requetes/AssignStatusTO.java]: This comment becomes obsolete

**Pull request merged**